src/Model/Behavior/SearchBehavior.php

```diff
@@ -30,6 +31,7 @@ class SearchBehavior extends Behavior
             'searchManager' => 'searchManager',
             'isSearch' => 'isSearch'
         ],
+        'emptyValues' => ['']
```

Comment:
Also you will have to ensure the array gets overwritten and not merged (which is default behavior) when custom value is set otherwise one would never be able to remove the default empty values.

Reply:
If we support CakePHP 3.3 we could add '0.0'.
- `Hash::filter()`: https://api.cakephp.org/3.3/source-class-Cake.Utility.Hash.html#645
- `empty()`: https://3v4l.org/sj8bM

Reply:
> Also you will have to ensure the array gets overwritten and not merged (which is default behavior) 
@ADmad actually how can I do that? I am not sure this part.

Reply:
@ADmad I realise that users can use configShallow to overwrite the config values